ssv/runner.go

```diff
@@ -167,30 +151,24 @@ func (b *BaseRunner) basePostConsensusMsgProcessing(runner Runner, signedMsg *ty
 	return hasQuorum, roots, errors.Wrap(err, "could not process post-consensus partial signature msg")
 }
 
-// basePartialSigMsgProcessing adds an already validated partial msg to the container, checks for quorum and returns true (and roots) if quorum exists
+// basePartialSigMsgProcessing adds an already validated partial msg to the container,
+// checks for quorum and returns true (and roots) if we should process the msg
 func (b *BaseRunner) basePartialSigMsgProcessing(
 	signedMsg *types.SignedPartialSignatureMessage,
-	container *PartialSigContainer,
+	container PartialSignatureContainer,
 ) (bool, [][32]byte, error) {
-	roots := make([][32]byte, 0)
-	anyQuorum := false
-	for _, msg := range signedMsg.Message.Messages {
-		prevQuorum := container.HasQuorum(msg.SigningRoot)
-
-		container.AddSignature(msg)
-
-		if prevQuorum {
-			continue
-		}
-
-		quorum := container.HasQuorum(msg.SigningRoot)
-		if quorum {
-			roots = append(roots, msg.SigningRoot)
-			anyQuorum = true
-		}
+
+	if b.Share.HasQuorum(len(container)) {
```

Comment:
it seems like `HasQuorum` just checks the number of sigs we have, don't we want to also check validity before we say we have quorum ?

Reply:
for a message to enter the container it should pass validity checks